BEIMA.Client/src/pages/DeviceTypes/AddDeviceTypePage.js

```diff
@@ -0,0 +1,129 @@
+import { Form, Card, Button, ListGroup } from 'react-bootstrap';
+import { useOutletContext } from 'react-router-dom';
+import { useState, useEffect } from "react";
+import { IoMdCloseCircle } from "react-icons/io";
+import styles from './AddDeviceTypePage.module.css';
+import FormList from '../../shared/FormList/FormList.js'
+
+const AddDeviceTypePage = () => {
+  // will be replaced with API call to get default fields
+  const defaultDeviceFields = [
+    "Building",
+    "Longitude",
+    "Latitude",
+    "Location Notes",
+    "Device Tag",
+    "Manufacturer",
+    "Model Number",
+    "Serial Number",
+    "Year Manufactured",
+    "Device Notes"
+  ]
+
+  // basic fields for all device types
+  const typeFields = {
+      "Name": "",
+      "Description": "",
+      "Device Type Notes": ""
+  }
+
+  const [deviceFields, setDeviceFields] = useState([]);
+  const [typeAttributes] = useState(typeFields);
+  const [setPageName] = useOutletContext();
+  let fullTypeJSON = {};
+  let field;
+
+  useEffect(() => {
+      setPageName('Add Device Type')
+  });
+
+  // allows the user to remove a field they added
+  function removeField(field) {
+    setDeviceFields(deviceFields.filter(item => item !== field))
+  }
+
+  // adds field to list
+  function addField(newField, event) {
+    let newList = deviceFields.concat(newField);
+    setDeviceFields(newList);
+    event.target.form.elements.newFieldForm.value = "";
+  }
+
+  // gathers input and puts it into JSON
+  function createJSON(event){
```

Comment:
Feels weird to use event here, maybe use something like "object"?

Reply:
Updated to use "object"

Reply:
On further inspection, it looks like this is actually meant to be an event object, maybe we call it something like addButtonEvent instead.

Reply:
Updated to addButtonEvent